_episodes/10-distribution.md

```diff
@@ -0,0 +1,245 @@
+---
+title: "Distributing Packages"
+teaching: 30
+exercises: 60
+questions:
+- "How can you get your project out there?"
+objectives:
+- "Discuss community strategies"
+keypoints:
+- ""
+---
+
+# Preliminary
+
+Register a test account on test.pypi.org.
+
+Create an access token, and *don't lose it!*
+(I keep a copy of the text for my token in my password manager.)
+
+Create `$HOME/.pypirc`. It should look something like the following.
+
+~~~
+[testpypi]
+  username = __token__
+  password = pypi-Abunchofbase64encodedbinarydataAbunchofbase64encodedbinarydataAbunchofbase64encodedbinarydataAbunchofbase64encodedbinarydata
+~~~
+{: .output}
+
+You will also eventually need to get an account on the main pypi.org,
+but you can do that later.
+
+Install some additional packages to help build and upload your distribution.
+~~~
+pip install build twine
+~~~
+{: .language-bash}
+
+# Preparing a release
+
+## Release notes
+
+Make sure your README file and documentation are up to date.
+Check the project description in your project's metadata, too.
+
+Consider whether there is some announcement text that you would
+want to include in a text file or in a blurb in your GitHub Releases tab.
+
+*TODO: Do we care about GitHub's Release and Tags tabs?*
+
+
+## Tagging the release
+
+~~~
+git status
+~~~
+{: .language-bash}
+
+Make sure that your changes are checked in and that you don't have
+extra files sitting around.
+
+Tag the latest commit with a version identifier.
+Let's get ready to publish `molecool` 1.0.0a1, the first alpha release of our package.
+~~~
+git tag 1.0.0a1
+~~~
+{: .language-bash}
+
+Build the distribution archive(s) for the packaging system you are using (see below): `PyPI` or `Conda`.
+
```

Comment:
```suggestion

~~~
Tags created in your local repository are not automatically shared when you `git push` a branch.
You can push tags the same way you push branches, or you can create the tag directly on GitHub.
With the GitHub interface, you can associate tags with
[Releases](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases)
to provide a place to publish release notes and automatically generate links to archive files of
your repository at the tagged commits.

Tags created through the GitHub interface (or pushed from a different location) will be available
locally the next time you `git fetch`.

Keep in mind that the GitHub repository page is one of the main entry points for new users of your package.
Your project looks healthier and better maintained with a nice, well documented history of
[releases](https://docs.github.com/en/repositories/releasing-projects-on-github/viewing-your-repositorys-releases-and-tags).
~~~
{: .callout}
```

Tags and GitHub releases do not automatically make your package easier to find or install,
but they can be helpful in preparing the source of version information or the source archive
as you prepare the *distribution* of your package.